File: src/main/java/com/gitlab/daring/image/command/structure/Contour.java
package com.gitlab.daring.image.command.structure;

import org.bytedeco.javacpp.opencv_core.Mat;

import java.util.EnumMap;

public class Contour {

	final Mat mat;
	final EnumMap<ContourMetric, Double> metrics = new EnumMap<>(ContourMetric.class);

	public Contour(Mat mat) {
		this.mat = mat;
	}

	public double getMetric(ContourMetric m) {
		return metrics.computeIfAbsent(m, mk -> m.calculate(mat));
	}

}

File: src/main/kotlin/com/gitlab/daring/image/command/structure/Contour.kt
package com.gitlab.daring.image.command.structure

import org.bytedeco.javacpp.opencv_core.Mat
import java.util.*

class Contour(val mat: Mat) {

    val metrics = EnumMap<ContourMetric, Double>(ContourMetric::class.java)

    fun getMetric(m: ContourMetric): Double {
        return metrics.computeIfAbsent(m) { m.calculate(mat) }
    }

}
